feat(components): mutations over time: add button to download the data

fengelniederhammer opened the pull request:
<!-- Mention the issue that this PR resolves. Delete if there is no corresponding issue -->
resolves #362

### Summary
<!-- 
Add a few sentences describing the main changes introduced in this PR
This is only relevant, if there is no issue that already contains the information
-->

I wasn't quite sure how to format the date ranges. I went for:
* days: `2024-01-20`
* weeks: `Week 3 2024`
* months: `2024-01`
* years: `2024`

### Screenshot
<!-- 
When applicable, add a screenshot showing the main effect this PR has, even if "trivial":
e.g. changed layout, changed button text, different logging in backend.
This helps others quickly grasping what you did even if they are not familiar with the code base.
-->
Day granularity: 
```
mutation,2024-01-20,2024-01-21,2024-01-22,2024-01-23,2024-01-24,2024-01-25,2024-01-26
ORF1a:T170I,0.2706422018348624,0.2868217054263566,0.2796934865900383,0.23921971252566734,0.2478839177750907,0.23929471032745592,0.27164685908319186
ORF1a:F499L,0.04308390022675737,0.06395348837209303,0.06782768102658111,0.06686626746506986,0.059718969555035126,0.05795314426633785,0.06451612903225806
...
``` 

Month granularity:
```
mutation,2024-01,2024-02,2024-03,2024-04,2024-05,2024-06,2024-07
C21T,0.927536231884058,0.9474178403755869,0.8532608695652174,0.7741935483870968,0.8264462809917356,0.7741935483870968,0
C44T,0.5814824057898678,0.5566378066378066,0.47790802524797116,0.5367647058823529,0.488929889298893,0.5367647058823529,0
C774T,0.25756390729839873,0.27434033916061024,0.2704632532218739,0.189280868385346,0.07071911005164878,0.189280868385346,0
C1762A,0.06653470146238992,0,0,0,0,0,0
C11747T,0.06460081773187003,0,0,0,0,0,0
G17562T,0.06467541852342527,0.050597916934550596,0,0,0,0,0
T18453C,0.20766863713661024,0.21222882231404958,0.210768444368717,0.12189466710831401,0,0.12189466710831401,0
G21641T,0.07507173933166712,0,0,0,0,0,0
T23011-,0.8804200680969767,0.9019581723724523,0.9095238095238095,0.9803122878479293,0.9950738916256158,0.9803122878479293,0

...
``` 

Week granularity:
```
mutation,Week 3 2024,Week 4 2024,Week 5 2024,Week 6 2024
C44T,0.583030303030303,0.6011976047904192,0.6339712918660287,0.5619524405506884
A518-,0.03741935483870968,0.03889255108767304,0.04767986377181779,0.05125258561250287
C774T,0.264943775892681,0.25326991676575505,0.27313201703445605,0.2669872464980138
C1762A,0.06457887570511574,0.06363810619129769,0.04821121101970537,0.04193481419970936
C11747T,0.0637101811367895,0.06081775220875282,0.04768611670020121,0.03982202447163515
G17562T,0.07279397567097895,0.07460408319023087,0.06580448839863066,0.05956306677658697
T18453C,0.21785644816183622,0.19922779922779923,0.2117737003058104,0.21245345469590401
...
``` 

### PR Checklist
<!-- Check completed items of strikethrough irrelevant items (using ~~text~~) -->
- [x] All necessary documentation has been adapted.
- [x] The implemented feature is covered by an appropriate test.

chaoran-chen commented:
ISO defines a format for weeks: e.g., `2024-W03`, I suggest going with that.

https://en.wikipedia.org/wiki/ISO_week_date

fengelniederhammer commented:
> ISO defines a format for weeks: e.g., `2024-W03`, I suggest going with that.
> 
> https://en.wikipedia.org/wiki/ISO_week_date

I changed it in a global manner. `YearWeek` now uses the format `'YYYY-[W]WW'` in its `toString`. That simplifies the logic in the mutations-over-time download button, but it also changed axis labels on other diagrams, such as prevalence over time. IMIO that is actually even better than the old format, because it's not ambiguous with months anymore.

The new downloaded files with months ganularity:
```
mutation,2024-W03,2024-W04,2024-W05,2024-W06
C44T,0.583030303030303,0.6011976047904192,0.6339712918660287,0.5619524405506884
A518-,0.03741935483870968,0.03889255108767304,0.04767986377181779,0.05125258561250287
C774T,0.264943775892681,0.25326991676575505,0.27313201703445605,0.2669872464980138
C1762A,0.06457887570511574,0.06363810619129769,0.04821121101970537,0.04193481419970936
C11747T,0.0637101811367895,0.06081775220875282,0.04768611670020121,0.03982202447163515
G17562T,0.07279397567097895,0.07460408319023087,0.06580448839863066,0.05956306677658697
T18453C,0.21785644816183622,0.19922779922779923,0.2117737003058104,0.21245345469590401
T21610C,0.024591947769314473,0.06403622250970246,0.02763018065887354,0.012640772236267524
...
```
